hoomd/mpcd/pytest/test_stream.py

```diff
@@ -0,0 +1,439 @@
+            np.testing.assert_array_almost_equal(
+                snap.mpcd.velocity, [[1.0, -1.0, 1.0], [-1.0, -1.0, -1.0]])
+
+        # take another step where one particle will now hit the wall
+        sim.run(1)
+        snap = sim.state.get_snapshot()
+        if snap.communicator.rank == 0:
+            np.testing.assert_array_almost_equal(
+                snap.mpcd.position, [[-4.85, 4.85, 3.95], [-0.2, -0.2, -4.0]])
+            np.testing.assert_array_almost_equal(
+                snap.mpcd.velocity, [[1.0, -1.0, -1.0], [-1.0, -1.0, -1.0]])
+
+        # take another step, wrapping the second particle through the boundary
+        sim.run(1)
+        snap = sim.state.get_snapshot()
+        if snap.communicator.rank == 0:
+            np.testing.assert_array_almost_equal(
+                snap.mpcd.position, [[-4.75, 4.75, 3.85], [-0.3, -0.3, -3.9]])
+            np.testing.assert_array_almost_equal(
+                snap.mpcd.velocity, [[1.0, -1.0, -1.0], [-1.0, -1.0, 1.0]])
+
+    def test_step_moving_wall(self, simulation_factory, snap):
+        """Test step with moving wall.
+
+        The first particle is matched exactly to the wall speed, and so it will
+        translate at same velocity along +x for 3 steps. It will bounce back in
+        y and z to where it started. (vx stays the same, and vy and vz flip.)
+
+        The second particle has y and z velocities flip again, and since it
+        started closer, it moves relative to original position.
+        """
+        if snap.communicator.rank == 0:
+            snap.mpcd.N = 2
+            snap.mpcd.position[:] = [[4.95, -4.95, 3.85], [0.0, 0.0, -3.8]]
+            snap.mpcd.velocity[:] = [[1.0, -1.0, 1.0], [-2.0, -1.0, -1.0]]
+        sim = simulation_factory(snap)
+        sm = hoomd.mpcd.stream.ParallelPlates(every=3, H=4, V=1, no_slip=True)
+        ig = hoomd.mpcd.Integrator(dt=0.1, streaming_method=sm)
+        sim.operations.integrator = ig
+
+        # run one step and check bounce back of particles
+        sim.run(1)
+        snap = sim.state.get_snapshot()
+        if snap.communicator.rank == 0:
+            np.testing.assert_array_almost_equal(
+                snap.mpcd.position, [[-4.75, -4.95, 3.85], [-0.4, -0.1, -3.9]])
+            np.testing.assert_array_almost_equal(
+                snap.mpcd.velocity, [[1.0, 1.0, -1.0], [0.0, 1.0, 1.0]])
+
+    def test_validate_box(self, simulation_factory, snap):
+        """Test box validation raises an error on run."""
+        sim = simulation_factory(snap)
+        sm = hoomd.mpcd.stream.ParallelPlates(every=1, H=10)
+        ig = hoomd.mpcd.Integrator(dt=0.1, streaming_method=sm)
+        sim.operations.integrator = ig
+
+        with pytest.raises(RuntimeError):
+            sim.run(1)
+
+    def test_test_of_bounds(self, simulation_factory, snap):
+        """Test box validation raises an error on run."""
+        if snap.communicator.rank == 0:
+            snap.mpcd.position[:] = [[4.95, -4.95, 3.85]]
```

Comment:
```suggestion
            snap.mpcd.position[0] = [[4.95, -4.95, 3.85]]
```
Only one particle anyways.